lockfiles/st2-constraints.txt

```diff
@@ -0,0 +1,68 @@
+# Add/remove version constraints for transitive dependencies in this file
+# (transitive dependencies are dependencies of our direct dependencies).
+# Then run `./pants generate-lockfiles --resolve=st2` to regenerate the lockfile.
+#
+# Direct dependencies should be recorded in `requirements-pants.txt`, not here.
+
+# ############################################ #
+# pinned transitive deps from requirements.txt #
+# ############################################ #
+#
+# required by jinja2:
+# fixed-requirements.txt:
+#     Fix MarkupSafe to < 2.1.0 as 2.1.0 removes soft_unicode
+#     >=0.23 was from jinja2
+MarkupSafe<2.1.0,>=0.23
```

Comment:
Because I'm not sure if they're still valid.

With `requirements-pants.txt`, I've essentially removed all of the versions that were "pinned"/"locked" in `fixed-requirements.txt`, because that is supposed to be handled by the lockfile. So, the requirements should be as broad as possible, and then the lockfile will lock them to a single version.

The same thing applies with `st2-constraints.txt`. I believe most of these transitive constraints were in `fixed-constraints.txt` in order to "pin"/"lock" them. But, some of them might be actual constraints due to known issues in those transitive deps. So, I documented all of them, but left them commented. I hope we can just delete them later.

As we enable other tools like pylint, we'll get additional feedback about whether or not we need some of these constraints. And, after we have the lockfile, we can start "exporting" a virtualenv with pants. That way we can run all of our tests using a virtualenv generated by the lockfile which will tell us if there are any functional issues where we still need these constraints.

Does that make sense?

I suppose we could also uncomment all of these constraints, and then comment one or more of them, if needed, to generate the lockfile.

Reply:
Another issue with our current approach vs using the pants+pex lockfile: generating the lockfile does not support the legacy pip resolver. Our current set of "pinned" dependencies does not work with the newer resolver (which is why we have a really old version of pip pinned), so we need to loosen our requirements to give the resolver room to figure out which versions are actually compatible.

I have successfully tested generating the lockfile with the requirements+constraints in this PR. I will try uncommenting the rest of these transitive dep constraints and see if that still resolves.

Reply:
OK. I uncommented many of the constraints and I can still generate the lockfile. I left only a few commented.

I really hope we can just delete all of these constraints, hopefully sooner rather than later.

Reply:
I just tried to cleanup / standardize the documentation of all of these constraints. @amanda11 does that help?

Reply:
Thanks for the answers. I seem to remember there being some problems going up further on the pip version - but that maybe resolved. But if we do end up upping the pip version might need to find the old PR from when I last upped the version - as I seem to recall having to downgrade it.
But agree for now good to keep in synch.